### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by davidhunglam**
- error handling
- naming

### CODE REVIEW

File: app/admin/webapp/extension/UploadDialog.fragment.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
                     id="uploader"
                     fileType="csv"
                     multiple="false"
-                    uploadUrl="/api/admin/Csv/data"
+                    uploadUrl="/api/admin/CsvStream/data"
```
Comment (beckermarc) [inline app/admin/webapp/extension/UploadDialog.fragment.xml:10]: From a URL perspective I think having something like `/api/admin/upload/csv` would be nice, but I guess any naming here would be fine for me.

File: srv/src/main/resources/messages_de.properties [hunk 2]
```diff
@@ -10,4 +10,5 @@ review.invalid.rating = Die Bewertung muss ein Wert innerhalb 1 und 5 sein
 review.added = Bewertung hinzugef\u00FCgt
 order.exceeds.stock = {0} ist mehr als f\u00FCr das Buch auf Vorrat \u00FCbrig ist
 review.add.forbidden=Es ist nicht mehr als eine Bewertung pro Buch erlaubt
-book.import.failed = B\u00FCcher Import fehlgeschlagen
\ No newline at end of file
+book.import.failed = B\u00FCcher Import fehlgeschlagen
+book.import.invalid.csv = Die CSV Datei enthält eine ungültige Struktur - Bitte \u00FCberpr\u00FCfen Sie den Inhalt
```
Comment (beckermarc) [inline srv/src/main/resources/messages_de.properties:14]: Do we still have issues with umlauts? The fact that you only replaced half of the umlauts maybe suggests that it is not required anymore to do this?

Comment (davidhunglam) [reply srv/src/main/resources/messages_de.properties:14]: No, I just forgot to change it when I tried out different phrases. I have not tested with German locale, I just assumed it works as suggested by the existing entries.

File: app/admin/webapp/extension/UploadDialog.fragment.xml (hunk 1 above)
Comment (davidhunglam) [reply app/admin/webapp/extension/UploadDialog.fragment.xml:10]: Upload sounds good, too, as we are not persisting the stream anyways. So there'd be no "download". But entities in the model have to start with a capital, right? Is there a way to override the entity name for the path?

File: srv/src/main/resources/messages_de.properties (hunk 2 above)
Comment (beckermarc) [reply srv/src/main/resources/messages_de.properties:14]: Okay :) I was always wondering if this is still necessary and just tested it and it is not. I think we fixed that bug long ago, that we couldn't handle umlauts there. I will open a separate PR. You can use normal umlauts for your new error texts.

File: app/admin/webapp/extension/UploadDialog.fragment.xml (hunk 1 above)
Comment (beckermarc) [reply app/admin/webapp/extension/UploadDialog.fragment.xml:10]: > But entities in the model have to start with a capital, right?

I don't think anybody enforces this, but I am not sure.

> Is there a way to override the entity name for the path only?

No currently not.

File: srv/src/main/resources/messages_de.properties (hunk 2 above)
Comment (davidhunglam) [reply srv/src/main/resources/messages_de.properties:14]: okay :)

File: app/admin/webapp/extension/UploadDialog.fragment.xml (hunk 1 above)
Comment (davidhunglam) [reply app/admin/webapp/extension/UploadDialog.fragment.xml:10]: > I don't think anybody enforces this, but I am not sure.

Actually, it works. The generated class will be annotated like this:
`@CdsName("AdminService.upload")`

But do we want to promote this?

File: srv/src/main/resources/messages_de.properties (hunk 2 above)
Comment (beckermarc) [reply srv/src/main/resources/messages_de.properties:14]: https://github.com/SAP-samples/cloud-cap-samples-java/pull/93

File: app/admin/webapp/extension/UploadDialog.fragment.xml (hunk 1 above)
Comment (beckermarc) [reply app/admin/webapp/extension/UploadDialog.fragment.xml:10]: > But do we want to promote this?

good question, not sure about that :)

Comment (davidhunglam) [reply app/admin/webapp/extension/UploadDialog.fragment.xml]: Playing it safe then :)

File: srv/src/main/java/my/bookshop/handlers/AdminServiceHandler.java [hunk 3]
```diff
@@ -261,9 +261,9 @@ public void addBookToOrder(AddToOrderContext context) {
 	/**
 	 * @return the static CSV singleton upload entity
 	 */
-	@On(entity = Csv_.CDS_NAME, event = CdsService.EVENT_READ)
-	public Csv getCsvSingleton() {
-		return Csv.create();
+	@On(entity = Upload_.CDS_NAME, event = CdsService.EVENT_READ)
+	public Upload getCsvSingleton() {
```
Comment (beckermarc) [inline srv/src/main/java/my/bookshop/handlers/AdminServiceHandler.java:265]:
```suggestion
	public Upload getUploadSingleton() {
```

**Review by beckermarc — APPROVED** [review]
LGTM

**Review by AndreasSchoesser — APPROVED** [review]

**Pull request merged**